lib/cc/services/github_pull_requests.rb

```diff
@@ -73,4 +89,82 @@ def response_includes_repo_scope?(response)
   def test_status_code
     422
   end
+
+  def welcome_comment_implemented?
+    true
+  end
+
+  def should_post_welcome_comment?
+    config.welcome_comment_enabled && @payload.fetch("first_contribution", false)
+  end
+
+  ALREADY_COMMENTED_PATTERN = /Since this is the first PR/
```

Comment:
I was more concerned about our retrying logic. I was thinking specifically if our status update fails, but the comment succeeds, we might retry and double-comment. With a little more context of the code, I actually don't think that will happen, because we'll fail hard on the status update failure. Hypothetically, if we were to add a third component of this service in the future, which takes place after the comment, and *that* were to fail, we might retry and double-comment (without this check).